libs/atqr/domain/src/lib/domain-spec/stories/visitorAdoptsGoal.disl

```diff
@@ -0,0 +1,28 @@
+read_also api.disl
+read_also enums.disl
+read_also ui.disl
+
+entity Goal
+  @goal: string
+  @challenges: Challenge[]
+  @pursuers: returns @challenges.map(c => c.enrollments.map(e => e.player)).flat()
+
+root Challenge
+  CreateChallenge?
+  AdoptGoal? :email :goal
```

Comment:
If you are creating a Challenge based on a Goal, a Challenge instance cannot be the one to handle the command since it does not exist yet. The goal does. So Goal will need to be another aggregate root, one capable of creating challenges. Btw, maybe all challenge creation should start on the goal (I have to think about that, what do you think?)

Reply:
It does make sense to me.The Goal will end up being kind of a birth place for of the all challenges.

Reply:
It does make sense to me.The Goal will end up being kind of a birth place for of the all challenges.